python/ray/serve/_private/router.py

```diff
@@ -305,6 +348,32 @@ def __init__(
             call_in_event_loop=event_loop,
         )
 
+        self.deployment_name = deployment_name
```

Comment:
no change in functionality. move logic to router

Reply:
1. Move Metrics pusher logic lifecycle out of the Router class.
2. Router will pass necessary arguments to initiate metrics pusher and start the pusher.